### PR TITLE
src/components: use merchandise data to render selection widget in FormFieldListMerch

### PR DESCRIPTION
**Pull request by maceksimon**
Use merchandise data to render selection widget in `FormFieldListMerch`.

For rendering data in UI we need:
- Composite data: Card (shows all available sizes for given product and gender)
- Individual options: Items with selectable IDs

Compute these values from `merchandise` data in the composable (in a lazy-load way).
Update displaying values in the template.
Update fixture so we can group items by name.

### CODE REVIEW

File: src/composables/useApiGetMerchandise.ts [hunk 1]
```diff
@@ -111,8 +126,205 @@ export const useApiGetMerchandise = (
...
+   * This is used to display merchandise cards with composite data
+   * showing information about all variants of the product of that name.
+   */
+  const computeMerchandiseCards = (): Record<Gender, MerchandiseCard[]> => {
+    logger?.info('Computing merchandise cards by gender');
+    logger?.debug(`Transform <${merchandise.value.length}> items to cards`);
+
+    // group merchandise by gender
+    const merchandiseGroupedByGender = merchandise.value.reduce(
+      (acc, item) => {
```
Comment (tmszi) [inline src/composables/useApiGetMerchandise.ts:229]: better var name `aac` (I don't know what does it mean 'aac')

File: src/composables/useApiGetMerchandise.ts [hunk 2]
```diff
@@ -111,8 +126,205 @@ export const useApiGetMerchandise = (
...
+        logger?.debug(
+          `Processing <${genderItems.length}> items for <${gender}>`,
+        );
+
+        /**
+         * Group merchandise by name within given gender
+         * This is done to distinguish between size options for each gender.
+         */
+        const merchandiseGroupedByName = genderItems.reduce(
+          (nameAcc, item) => {
```
Comment (tmszi) [inline src/composables/useApiGetMerchandise.ts:255]: Same as my comment above `nameAcc`.

Comment (tmszi) [reply src/composables/useApiGetMerchandise.ts:255]: This function `(nameAcc, item) => {` repeat see line 229, please.

File: src/composables/useApiGetMerchandise.ts [hunk 3]
```diff
@@ -111,8 +126,205 @@ export const useApiGetMerchandise = (
...
+          },
+          {} as Record<string, Merchandise[]>,
+        );
+
+        logger?.debug(
+          `Items by name <${JSON.stringify(merchandiseGroupedByName, null, 2)}>`,
+        );
+
+        // transform each name group into MerchandiseCard
+        acc[gender as Gender] = Object.entries(merchandiseGroupedByName).map(
```
Comment (tmszi) [inline src/composables/useApiGetMerchandise.ts:270]: unreadable var name `aac`, see my comment above.

File: src/composables/useApiGetMerchandise.ts [hunk 4]
```diff
@@ -111,8 +126,205 @@ export const useApiGetMerchandise = (
     }
   };
 
+  const merchandiseGroupedByName = computed(() => {
+    return merchandise.value.reduce(
+      (acc, item) => {
```
Comment (tmszi) [inline src/composables/useApiGetMerchandise.ts:131]: Same as my comments bellow.

File: src/composables/useApiGetMerchandise.ts (hunk 1 above)
Comment (maceksimon) [reply src/composables/useApiGetMerchandise.ts:229]: Yes, `acc` is an abbreviation for "accumulated value", but the name should be clearer.
Renamed to `groups` + refactored the reduce function to standalone `groupMerchandiseByKey` function (based on later comment).

File: src/composables/useApiGetMerchandise.ts (hunk 2 above)
Comment (maceksimon) [reply src/composables/useApiGetMerchandise.ts:255]: Renamed to `groups` + refactored the reduce function to standalone `groupMerchandiseByKey` function (based on later comment).

File: src/composables/useApiGetMerchandise.ts (hunk 3 above)
Comment (maceksimon) [reply src/composables/useApiGetMerchandise.ts:270]: Renamed the accummulated value to `genderGroupsContainingNameGroups`.
Renamed the `merchandiseGroupedByName` variable to more precise `merchandiseWithCurrentGenderGroupedByName`

File: src/composables/useApiGetMerchandise.ts (hunk 4 above)
Comment (maceksimon) [reply src/composables/useApiGetMerchandise.ts:131]: Renamed to groups + refactored the reduce function to standalone `groupMerchandiseByKey` function (based on your other comment).

**Review by tmszi — APPROVED** [review]

**Pull request merged**